src/components/sbb-timetable-row/sbb-timetable-row.helper.tsx

```diff
@@ -96,3 +106,94 @@ export const renderStringProduct = (vehicleName: string, line: string): JSX.Elem
     </span>
   );
 };
+
+const isReachable = (legs: PTRideLeg[]): boolean => {
+  return legs?.some((leg) => leg.serviceJourney?.serviceAlteration?.reachable === true);
+};
+
+const isRedirected = (legs: PTRideLeg[]): boolean => {
+  return legs?.some((leg) => leg.serviceJourney?.serviceAlteration?.redirected === true);
+};
+
+const isUnplannedStop = (legs: PTRideLeg[]): boolean => {
+  return legs?.some(
+    (leg) => !!leg.serviceJourney?.serviceAlteration?.unplannedStopPointsText === true
+  );
+};
+
+export const sortSituation = (situations: PtSituation[]): PtSituation[] => {
+  const priorities = {
+    DISTURBANCE: 0,
+    INFORMATION: 1,
+    DELAY: 2,
+    TRAIN_REPLACEMENT_BY_BUS: 3,
+    CONSTRUCTION_SITE: 4,
+    END_MESSAGE: 5,
+  };
+
+  return situations
+    ?.slice()
+    ?.sort((a: PtSituation, b: PtSituation) => priorities[a?.cause] - priorities[b?.cause]);
+};
+
+export const getHimIcon = (situation: PtSituation): string => {
+  switch (situation?.cause) {
+    case 'DISTURBANCE':
+      return 'disruption';
+    case 'INFORMATION':
+      return 'info';
+    case 'DELAY':
+      return 'delay';
+    case 'TRAIN_REPLACEMENT_BY_BUS':
+      return 'replacementbus';
+    case 'CONSTRUCTION_SITE':
+      return 'construction';
+    default:
+      return 'info';
+  }
+};
+
+export const getCus = (trip: Trip): string[] => {
+  const cus = [];
+  (trip?.summary.tripStatus?.cancelled || trip?.summary.tripStatus?.partiallyCancelled) &&
```

Comment:
This is hard to read and has too much redundancy. 
Lets have a quick call on how to clean this up :-)